src/main/java/seedu/address/model/feed/Feed.java

```diff
@@ -37,46 +43,114 @@ public Feed(String name, String address) {
      * @return List of maximum 5 posts.
      */
     public ObservableList<FeedPost> fetchPosts() {
-        ObservableList<FeedPost> posts = FXCollections.observableArrayList();
+        String feedData = this.fetchFeedData();
+        ObservableList<FeedPost> feedPosts = this.parseFeedData(feedData);
+
+        return feedPosts;
+    }
+
+    /**
+     * Fetches the feed's data as a String. This method first attempts to fetch data from the feed's remote
+     * address. If unsuccessful, it attempts to read the local cached feed data.
+     * @return String representing the feed data.
+     */
+    private String fetchFeedData() {
+        String feedData = null;
+        String xmlFile = String.format("data/cache/%s.xml", name.toLowerCase().replaceAll(" ", "-"));
 
         try {
+            // Fetch remote
             URL url = new URL(address);
             HttpURLConnection conn = (HttpURLConnection) url.openConnection();
             BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
 
+            int c;
+            StringBuilder buffer = new StringBuilder();
+
+            while ((c = in.read()) != -1) {
+                buffer.append((char) c);
+            }
+            feedData = buffer.toString();
```

Comment:
does this mean you are reading the entire feed of the website?

Reply:
Updated this to read only the necessary amount